crates/libs/windows/src/core/factory_cache.rs

```diff
@@ -98,18 +103,14 @@ pub fn factory<C: RuntimeName, I: Interface>() -> Result<I> {
             library[..path.len()].copy_from_slice(path.as_bytes());
             library[path.len()..].copy_from_slice(b".dll\0");
 
-            let function = delay_load(library, b"DllGetActivationFactory\0");
-
             if !function.is_null() {
-                let function: DllGetActivationFactory = core::mem::transmute(function);
-                let mut abi = core::ptr::null_mut();
-                let _ = function(core::mem::transmute_copy(&name), &mut abi);
+                let result = get_activation_factory(library, &name);
```

Comment:
```suggestion
                if let Ok(factory) = get_activation_factory(library, &name) {
                    return factory.cast();
                }
```

Reply:
Note also that this code will never run since the `function` points to the wrong function.

Reply:
Actually, that's not entirely true by the looks of it but that `is_null` check is still suspect.